staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -102,22 +108,39 @@ func NewNamedReflector(name string, lw ListerWatcher, expectedType interface{},
 		name:          name,
 		listerWatcher: lw,
 		store:         store,
-		expectedType:  reflect.TypeOf(expectedType),
 		period:        time.Second,
 		resyncPeriod:  resyncPeriod,
 		clock:         &clock.RealClock{},
 	}
+	r.setExpectedType(expectedType)
 	return r
 }
 
+func (r *Reflector) setExpectedType(expectedType interface{}) {
+	r.expectedTypeName = "<unknown>"
```

Comment:
it's more like `<unspecified>` than unknown.

Reply:
Done